plugins/importexport/medra/MedraExportPlugin.inc.php

```diff
@@ -131,74 +135,42 @@ function depositXML($objects, $context, $filename) {
-		                    echo "<td style='border:1px solid black; font-size:18; font-weight:bold; padding:10px;'>DESCRIZIONE</td>";
-		                } else {
-		                    echo "<td style='border:1px solid black; font-size:18; font-weight:bold; padding:10px;'>CODE</td>";
-		                    echo "<td style='border:1px solid black; font-size:18; font-weight:bold; padding:10px;'>ELEMENT</td>";
-		                    echo "<td style='border:1px solid black; font-size:18; font-weight:bold; padding:10px;'>DESCRIPTION</td>";
-		                }
-		                
-		                echo '</tr>';
-		            }
-		            
 		            $nodeList = $doc->getElementsByTagName('error');
-		            $length = $nodeList->length;
-		            for ($i = 0; $i < $length; $i++) {
-		                
-		                $doc->getElementsByTagName('code')->item($i)->textContent;
-		                if($i % 2 == 0){
-		                    echo "<tr style='background-color: #ABCDEF;'>";
-		                } else {
-		                    echo "<tr style='background-color: #BDB9AA;'>";
+		            
+		            $headlines = array();
+		            
+		            foreach($nodeList as $node) {
+		                $headline = array();
+		                if($node->childNodes->length) {
+		                    foreach($node->childNodes as $i) {
+		                        $headline[$i->nodeName] = $i->nodeValue;
+		                    }
 		                }
 		                
-		                echo "<td style='border:1px solid black; font-size:18; padding:10px;'>" .
-		  		                '<a style="text-decoration:none;" href="' . substr($endpoint, 0, strrpos($endpoint, "servlet")) . 'en/crossrefErrorList.htm#' . $doc->getElementsByTagName('code')->item($i)->textContent . '">' .
-		  		                $doc->getElementsByTagName('code')->item($i)->textContent  .
-		  		                '</a>' .
-		  		                '</td>';
-		  		                echo "<td style='border:1px solid black; font-size:18; padding:10px;'>" . $doc->getElementsByTagName('reference')->item($i)->textContent  . '</td>';
-		  		                echo "<td style='border:1px solid black; font-size:18; padding:10px;'>" . $doc->getElementsByTagName('description')->item($i)->textContent  . '</td>';
-		  		                echo '</tr>';
-		  		                
+		                $headlines[] = $headline;
 		            }
 		            
-		            if($numberError > 0){
-		                echo '</table>';
-		            }
+		            $templateMgr->assign(
+		                'headlines', $headlines
+		            );
 		            
 		        }
 		        
-		        libxml_clear_errors();
-		        echo '<h3>' . __('plugins.importexport.common.invalidXML') . '</h3>';
-		        echo '<p><pre>' . htmlspecialchars($xml) . '</pre></p>';
-		        echo '</body></html>';
+		        $templateMgr->assign(
+		            'htmlspecialchars', htmlspecialchars($xml)
```

Comment:
You can use an array syntax for the `assign` function wen assigning several variables in a row:
$templateMgr->assign([
  'variable1' => $variable1,
  'variable2' => $variable2,
]);

Reply:
Could you give please some suggestions? any tips?

Reply:
In this example, you can replace
```
$templateMgr->assign(
    'htmlspecialchars', htmlspecialchars($xml)
);

$templateMgr->assign(
    'numberError', $numberError
);
```
...with...
```
$templateMgr->assign([
    'htmlspecialchars' => htmlspecialchars($xml),
    'numberError' => $numberError,
]);
```
It's shorter and more readable IMO.